access/resource_table_acl.go

```diff
@@ -0,0 +1,351 @@
+				ta.Denies = append(ta.Denies, tablePerms)
+				privileges = &ta.Denies[len(ta.Denies)-1].Privileges
+			} else {
+				ta.Grants = append(ta.Grants, tablePerms)
+				privileges = &ta.Grants[len(ta.Grants)-1].Privileges
+			}
+		}
+		*privileges = append(*privileges, actionType)
+	}
+	return nil
+}
+
+func (ta *TableACL) revoke() error {
+	existing, err := loadTableACL(ta.ID())
+	if err != nil {
+		return err
+	}
+	existing.exec = ta.exec
+	existing.ClusterID = ta.ClusterID
+	if err = existing.read(); err != nil {
+		return err
+	}
+	for _, tps := range existing.permissions() {
+		for _, priv := range tps {
+			if err = ta.apply(func(objType, key string) string {
+				return fmt.Sprintf("REVOKE ALL PRIVILEGES ON %s %s FROM `%s`",
+					objType, key, priv.Principal)
+			}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (ta *TableACL) enforce() (err error) {
+	if err = ta.revoke(); err != nil {
+		return err
+	}
+	for action, grantsOrDenies := range ta.permissions() {
+		for _, grant := range grantsOrDenies {
+			if err = ta.apply(func(objType, key string) string {
+				privileges := strings.Join(grant.Privileges, ", ")
+				return fmt.Sprintf("%s %s ON %s %s TO `%s`",
+					action, privileges, objType, key, grant.Principal)
+			}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (ta *TableACL) apply(qb func(objType, key string) string) error {
+	sqlQuery := qb(ta.TypeAndKey())
+	log.Printf("[INFO] Executing SQL: %s", sqlQuery)
+	r := ta.exec.Execute(ta.ClusterID, "sql", sqlQuery)
+	return r.Err()
+}
+
+func (ta *TableACL) initCluster(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) (err error) {
+	clustersAPI := compute.NewClustersAPI(ctx, c)
+	if ci, ok := d.GetOk("cluster_id"); ok {
+		ta.ClusterID = ci.(string)
+	} else {
+		ta.ClusterID, err = ta.getOrCreateCluster(clustersAPI)
+		if err != nil {
+			return
+		}
+	}
+	clusterInfo, err := clustersAPI.StartAndGetInfo(ta.ClusterID)
+	if err != nil {
+		return
+	}
+	if v, ok := clusterInfo.SparkConf["spark.databricks.acl.dfAclsEnabled"]; !ok || v != "true" {
+		err = fmt.Errorf("cluster_id: not a High-Concurrency cluster: %s (%s)",
+			clusterInfo.ClusterName, clusterInfo.ClusterID)
+		return
+	}
+	ta.exec = c.CommandExecutor(ctx)
+	return nil
+}
+
+func (ta *TableACL) getOrCreateCluster(clustersAPI compute.ClustersAPI) (string, error) {
+	sparkVersion := clustersAPI.LatestSparkVersionOrDefault(compute.SparkVersionRequest{
+		Latest: true,
+	})
+	nodeType := clustersAPI.GetSmallestNodeType(compute.NodeTypeRequest{LocalDisk: true})
+	aclCluster, err := clustersAPI.GetOrCreateRunningCluster("terrraform-table-acl", compute.Cluster{
+		ClusterName:            "terrraform-table-acl",
+		SparkVersion:           sparkVersion,
+		NodeTypeID:             nodeType,
+		AutoterminationMinutes: 10,
+		SparkConf: map[string]string{
+			"spark.databricks.acl.dfAclsEnabled": "true",
+			"spark.master":                       "local[*]",
```

Comment:
I think that you also need: `"spark.databricks.cluster.profile" : "singleNode"` in the `SparkConf`: https://github.com/databrickslabs/terraform-provider-databricks/blob/master/storage/mounts.go#L126